pkg/services/ssosettings/api/api_test.go

```diff
@@ -1,3 +1,149 @@
 package api
 
-// TODO: add tests when you implement the final version of the API endpoint
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/accesscontrol/acimpl"
+	"github.com/grafana/grafana/pkg/services/org"
+	"github.com/grafana/grafana/pkg/services/ssosettings"
+	"github.com/grafana/grafana/pkg/services/ssosettings/ssosettingstests"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/web/webtest"
+)
+
+func TestSSOSettingsAPI_Delete(t *testing.T) {
+	type TestCase struct {
+		desc                string
+		key                 string
+		action              string
+		scope               string
+		expectedError       error
+		expectedServiceCall bool
+		expectedStatusCode  int
+	}
+
+	tests := []TestCase{
+		{
+			desc:                "successfully deletes SSO settings",
+			key:                 "azuread",
+			action:              "settings:write",
+			scope:               "settings:auth.azuread:*",
```

Comment:
Good point. For the delete method the user needs the wildcard scope because the delete method cannot remove specific config settings for a provider. It can only delete all settings associated with a provider.

For update I think we can optimize the required scope because the API can do partial updates.

@mgyongyosi what do you think?

Reply:
> do we also allow updating one specific setting for a specific auth provider?
No, we don't support this right now. There is a possibility to add it later.

Reply:
Gotcha, thanks for responding. Makes sense not to have it in the first iteration (and maybe at all).